gems/smithy/lib/smithy/templates/client/protocol_spec.erb

```diff
@@ -0,0 +1,164 @@
+# frozen_string_literal: true
+
+# This is generated code!
+
+require_relative '../spec_helper'
+<% additional_requires.each do |r| -%>
+require '<%= r %>'
+<% end -%>
+
+
+module <%= module_name %>
+  # TODO: Can be replaced by stub_responses config once implemented
+  class StubSend < Smithy::Client::Plugin
+    option(:stub_response)
+    handle(step: :send) do |context|
+      if (stub_response = context.config.stub_response)
+        resp = context.response
+        resp.signal_headers(stub_response[:status_code], stub_response.fetch(:headers, {}))
+        resp.signal_data(stub_response[:body]) if stub_response[:body]
+        resp.signal_done
+      end
+      Smithy::Client::Output.new(context: context)
+    end
+  end
+
+  describe Client do
+    before(:all) { Client.add_plugin(StubSend) }
+    after(:all) { Client.remove_plugin(StubSend) }
+
+    let(:client_options) do
+      {
+        # stub_responses: true,
+        # validate_input: false,
+        endpoint: 'http://127.0.0.1',
+        # retry_strategy: # disable?
+      }
+    end
+
+    let(:client) { Client.new(client_options) }
+
+<% all_operation_tests.each do |operation_tests| %>
+    describe '#<%= operation_tests.name %>' do
+<% unless operation_tests.request_tests.empty? -%>
+      describe 'requests' do
+<% operation_tests.request_tests.each do |test| %>
+<% test.comments.each do |line| -%>
+        # <%= line %>
+<% end -%>
+<% if test.skip? -%>
+        it '<%= test.id %>', skip: '<%= test.skip_reason %>' do
+<% else -%>
+        it '<%= test.id %>' do
+<% end -%>
+<% if test['host'] -%>
+          client = Client.new(client_options.merge(endpoint: '<%= test.test_case.fetch('authScheme', 'http') %>://<%= test['host'] %>'))
+<% end -%>
+<% if test.idempotency_token_trait? -%>
+          allow(SecureRandom).to receive(:uuid).and_return('00000000-0000-4000-8000-000000000000')
```

Comment:
Is there a better way to match expectations here? This is very implementation specific. Do the protocol tests also force this value?

Reply:
Prior art from Hearth (and I think v3).

The protocol tests force and expect this value.

Reply:
A matcher could say we make a uuid that's "like" the one we expect. It's not a strong assertion though. This is otherwise always saying we use secure random as an implementation which is probably fine.